rosbag2_cpp/include/rosbag2_cpp/reshapers/stitcher.hpp

```diff
@@ -0,0 +1,156 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// Copyright 2021 Firefly Automatix, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_CPP__RESHAPERS__STITCHER_HPP_
+#define ROSBAG2_CPP__RESHAPERS__STITCHER_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rosbag2_cpp/readers/sequential_reader.hpp"
+#include "rosbag2_cpp/writers/sequential_writer.hpp"
+
+// This is necessary because of using stl types here. It is completely safe, because
+// a) the member is not accessible from the outside
+// b) there are no inline functions.
+#ifdef _WIN32
+# pragma warning(push)
+# pragma warning(disable:4251)
+#endif
+
+namespace rosbag2_cpp
+{
+namespace reader_interfaces
+{
+class BaseReaderInterface;
+}  // namespace reader_interfaces
+namespace writer_interfaces
+{
+class BaseWriterInterface;
+}  // namespace writer_interfaces
+
+class ROSBAG2_CPP_PUBLIC Stitcher final
```

Comment:
Docstring for all classes please. What _exactly_ is the intended purpose of this class? That helps inform all following review